node/src/config_files/p2p.rs

```diff
@@ -112,6 +113,8 @@ impl From<P2pConfigFile> for P2pConfig {
             outbound_connection_timeout: c.outbound_connection_timeout.into(),
             mdns_config: mdns_config.into(),
             request_timeout: c.request_timeout.map(Duration::from_secs).into(),
+            // TODO: This should probably be set indirectly, for example by the node type.
+            announcement_subscriptions: AnnouncementSubscriptions::default(),
```

Comment:
I don't think having announcement subscriptions are the correct thing to do. This is too low-level. The right thing to do is to define multiple sets of possible subscriptions with node types. Currently, we can have one node type, the full node, which subscribes to whatever is available now. There's another type that we can support, which is the "BlocksOnly" mode for the sake of variety to be able to test it, at least. Please see how bitcoin does this. The entry point is here:

https://github.com/bitcoin/bitcoin/blob/9e229a542ff2107be43eff2e4b992841367f0366/src/init.cpp#L419